azure/services/agentpools/spec.go

```diff
@@ -201,6 +216,10 @@ func (s *AgentPoolSpec) Parameters(ctx context.Context, existing *asocontainerse
 	agentPool.Spec.EnableFIPS = s.EnableFIPS
 	agentPool.Spec.EnableEncryptionAtHost = s.EnableEncryptionAtHost
 
+	if kubernetesVersion := s.GetManagedMachinePoolVersion(existing); kubernetesVersion != nil {
+		agentPool.Spec.OrchestratorVersion = kubernetesVersion
+	}
```

Comment:
The only reason why i kept that check is, AKS allows creation of kubernetes clusters without a version. If no version is specified it picks the stable k8s version. In future if we would want to support it we will have to make this check and it would not be redundant. But yes at this point of time it is redundant as we don't allow k8s version to be nil.